packages/json-rpc-engine/src/v2/utils.ts

```diff
@@ -33,9 +34,29 @@ export function stringify(value: unknown): string {
   return JSON.stringify(value, null, 2);
 }
 
+const JsonRpcEngineErrorSymbol = Symbol.for('JsonRpcEngineError');
+
 export class JsonRpcEngineError extends Error {
+  private readonly [JsonRpcEngineErrorSymbol] = true;
+
   constructor(message: string) {
     super(message);
     this.name = 'JsonRpcEngineError';
   }
+
+  /**
+   * Check if a value is a {@link JsonRpcEngineError} instance.
+   * Works across different package versions in the same realm.
+   *
+   * @param value - The value to check.
+   * @returns Whether the value is a {@link JsonRpcEngineError} instance.
+   */
+  static isInstance<Value extends Error>(
```

Comment:
Interesting. Is there a case where you needed this?

Reply:
I have been in the case where I want to know if something is a `JsonRpcError` and `instanceof` has not availed me due to multiple different versions at play. I could see the same thing happening with the new `JsonRpcEngineError`, since we have multiple versions of `JsonRpcEngine` in production at the moment (e.g. [3 different major versions in the extension](https://github.com/MetaMask/metamask-extension/blob/9d9e625b37d10ba57615257cbe117eaf770c6710/yarn.lock#L6480-L6511)).

Edit: It's possible that those different versions do not occur in the same realm, but if they ever do, this ensures that there won't be a problem.